src/vdaf/mastic.rs

```diff
@@ -379,32 +382,418 @@ where
+                    .level_and_prefixes
+                    .prefixes()
+                    .len()
+        ]);
+        for output_share in output_shares.into_iter() {
+            agg_share.accumulate(&output_share)?;
+        }
+        Ok(agg_share)
+    }
+}
+
+impl<T, P, const SEED_SIZE: usize> Collector for Mastic<T, P, SEED_SIZE>
+where
+    T: Type,
+    P: Xof<SEED_SIZE>,
+{
+    fn unshard<M: IntoIterator<Item = Self::AggregateShare>>(
+        &self,
+        agg_param: &MasticAggregationParam,
+        agg_shares: M,
+        _num_measurements: usize,
+    ) -> Result<Self::AggregateResult, VdafError> {
+        let n = agg_param.level_and_prefixes.prefixes().len();
+        let mut agg_final = MasticAggregateShare::<T::Field>::from(vec![
+            T::Field::zero();
+            self.vidpf.weight_parameter
+                * n
+        ]);
+        for agg_share in agg_shares.into_iter() {
+            agg_final.merge(&agg_share)?;
+        }
+        let mut result = Vec::<T::AggregateResult>::with_capacity(n);
+        for i in 0..n {
+            let encoded_result = &agg_final.0
+                [i * self.vidpf.weight_parameter..(i + 1) * self.vidpf.weight_parameter];
+            result.push(
+                self.szk
+                    .typ
+                    .decode_result(&self.szk.typ.truncate(encoded_result.to_vec())?[..], 1)?,
+            );
+        }
+        Ok(result)
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
     use crate::field::Field128;
     use crate::flp::gadgets::{Mul, ParallelSum};
     use crate::flp::types::{Count, Sum, SumVec};
+    use crate::vdaf::test_utils::{run_vdaf, run_vdaf_prepare};
     use rand::{thread_rng, Rng};
 
     const TEST_NONCE_SIZE: usize = 16;
 
     #[test]
-    fn test_mastic_shard_sum() {
+    fn test_mastic_sum() {
         let algorithm_id = 6;
         let sum_typ = Sum::<Field128>::new(5).unwrap();
-        let sum_szk = Szk::new_turboshake128(sum_typ, algorithm_id);
         let sum_vidpf = Vidpf::<VidpfWeight<Field128>, TEST_NONCE_SIZE>::new(5);
-
+        let sum_szk = Szk::new_turboshake128(sum_typ, algorithm_id);
         let mut nonce = [0u8; 16];
         let mut verify_key = [0u8; 16];
         thread_rng().fill(&mut verify_key[..]);
         thread_rng().fill(&mut nonce[..]);
 
-        let first_input = VidpfInput::from_bytes(&[15u8, 0u8, 1u8, 4u8][..]);
-
+        let first_input = VidpfInput::from_bytes(&[240u8, 0u8, 1u8, 4u8][..]);
+        let second_input = VidpfInput::from_bytes(&[112u8, 0u8, 1u8, 4u8][..]);
+        let third_input = VidpfInput::from_bytes(&[48u8, 0u8, 1u8, 4u8][..]);
+        let fourth_input = VidpfInput::from_bytes(&[32u8, 0u8, 1u8, 4u8][..]);
+        let fifth_input = VidpfInput::from_bytes(&[0u8, 0u8, 1u8, 4u8][..]);
+        let first_prefix = VidpfInput::from_bools(&[false, false, true]);
+        let second_prefix = VidpfInput::from_bools(&[false]);
+        let third_prefix = VidpfInput::from_bools(&[true]);
```

Comment:
Put this in an array and reference the indices of the array below.
```suggestion
        let prefixes = [
            VidpfInput::from_bytes(&[240u8, 0u8, 1u8, 4u8][..]),
            VidpfInput::from_bytes(&[112u8, 0u8, 1u8, 4u8][..]),
            ... and so on
        ];
```